system/Commands/Database/ShowTableInfo.php

```diff
@@ -0,0 +1,236 @@
+    ];
+
+    public function run(array $params)
+    {
+        $this->db = Database::connect();
+
+        $tables = $this->db->listTables();
+
+        $this->sortIsDESC = CLI::getOption('desc');
+
+        if ($tables === []) {
+            CLI::error('Database has no tables!', 'light_gray', 'red');
+            CLI:newLine();
+
+            return;
+        }
+
+        if (CLI::getOption('show')) {
+            CLI::write('The following is a list of the names of all database tables : ', 'black', 'yellow');
+            CLI::newLine();
+
+            $thead = ['#ID', 'table name', 'number of rows', 'number of field'];
+
+            $tbody = $this->makeTbodyForShowAllTables($tables);
+
+            CLI::table($tbody, $thead);
+            CLI::newLine();
+
+            return;
+        }
+
+        $tableName = array_shift($params);
+
+        if (in_array($tableName, $tables, true)) {
+            if (CLI::getOption('metadata')) {
+                $this->showFieldMetaData($tableName);
+
+                return;
+            }
+
+            CLI::newLine();
+            CLI::write("Data of table \"{$tableName}\":", 'black', 'yellow');
+            CLI::newLine();
+
+            $thead = $this->db->getFieldNames($tableName);
+            $tbody = $this->makeTableRows($tableName);
+            CLI::table($tbody, $thead);
+
+            return;
+        }
+
+        $tableName = CLI::promptByKey(['Here is the list of your database tables:', 'Which table do you want to see?'], $tables, 'required');
+
+        if (CLI::getOption('metadata')) {
+            $this->showFieldMetaData($tables[$tableName]);
+
+            return;
+        }
+
+        CLI::newLine();
+        CLI::write("Data of table \"{$tables[$tableName]}\":", 'black', 'yellow');
+        CLI::newLine();
+
+        $thead = $this->db->getFieldNames($tables[$tableName]);
+
+        $tbody = $this->makeTableRows($tables[$tableName]);
+
+        CLI::table($tbody, $thead);
+    }
+
+    protected function makeTbodyForShowAllTables(array $tables): array
+    {
+        foreach ($tables  as $id => $tableName) {
+            $db = $this->db->query("SELECT * FROM {$tableName}");
+
+            $tbody[] = [
+                $id + 1,
+                $tableName,
+                $db->getNumRows(),
+                $db->getFieldCount(),
+
+            ];
+        }
+
+        if ($this->sortIsDESC) {
+            krsort($tbody);
+        }
+
+        return $tbody;
+    }
+
+    protected function makeTableRows(string $tableName): array
+    {
+        $limitRows = (int) CLI::getOption('limit-rows');
+        if (in_array($limitRows, [null, true, 0, 1], true)) {
+            // Default Value
+            $limitRows = 10;
+        }
+
+        $limitFieldsValue = (int) CLI::getOption('limit-fields-value');
+        if (in_array($limitFieldsValue, [null, true, 0, 1], true)) {
+            // Default Value
+            $limitFieldsValue = 15;
+        }
```

Comment:
Same comment above on casting to int.
```suggestion
        $limitFieldsValue = (int) CLI::getOption('limit-fields-value');

        if (in_array($limitFieldsValue, [null, true, 0, 1], true)) {
            $limitFieldsValue = 15;
        }
```